File: 121.best-time-to-buy-and-sell-stock.cpp
#include <bits/stdc++.h>
using namespace std;
/*
 * @lc app=leetcode id=121 lang=cpp
 *
 * [121] Best Time to Buy and Sell Stock
 */

// @lc code=start
class Solution
{
public:
    int maxProfit(vector<int> &prices)
    {
        int maxp = 0;
        int minp = INT_MAX;

        for (int i = 0; i < prices.size(); i++)
        {
            if (prices[i] < minp)
            {
                minp = prices[i];
            }
            else if (prices[i] - minp > maxp)
            {
                maxp = prices[i] - minp ;
            }
        }
        return maxp;
    }
};
// @lc code=end

File: Java/Daily Problems/February2023/121.best-time-to-buy-and-sell-stock.java
package February2023;
/*
 * @lc app=leetcode id=121 lang=java
 *
 * [121] Best Time to Buy and Sell Stock
 */

// @lc code=start
class Solution {
    public int maxProfit(int[] prices) {
        int minPrice = Integer.MAX_VALUE, profit = 0;
        for (int i = 0; i < prices.length; i++) {
            if (prices[i] < minPrice) {
                minPrice = prices[i];
                continue;
            }
            profit = Math.max(profit, prices[i] - minPrice);
        }
        return profit;
    }
}
// @lc code=end